pkgs/os-specific/linux/nixos-rebuild/nixos-rebuild.sh

```diff
@@ -653,18 +653,41 @@ fi
 # If we're not just building, then make the new configuration the boot
 # default and/or activate it now.
 if [[ "$action" = switch || "$action" = boot || "$action" = test || "$action" = dry-activate ]]; then
+    # Using systemd-run here to protect against PTY failures/network
+    # disconnections during rebuild.
+    # See: https://github.com/NixOS/nixpkgs/issues/39118
+    cmd=(
+        "systemd-run"
+        "-E" "LOCALE_ARCHIVE"
+        "-E" "NIX_PATH"
+        "-E" "PATH"
```

Comment:
This value comes from the old system, not the new system. I wonder if that's intentional...
(but not a regression fwiw)

Reply:
I think this is also not actually needed, let me double check.

Reply:
Yes, not needed. Will remove it too.